LibWeb: Don't play initially-paused animations

Gingeh opened the pull request:
also appears to gain a few passing subtests, yippee.

I changed `Tests/LibWeb/Text/input/WebAnimations/misc/animation-events-basic.html` to play the animation before waiting because the previously expected behaviour (running despite being paused) was incorrect.

tcl3 commented on Libraries/LibWeb/DOM/Element.cpp:
```diff
@@ -3968,9 +3968,6 @@ void Element::play_or_cancel_animations_after_display_property_change()
     auto play_or_cancel_depending_on_display = [&](Animations::Animation& animation) {
         if (has_display_none_inclusive_ancestor) {
             animation.cancel();
-        } else {
-            HTML::TemporaryExecutionContext context(realm());
-            animation.play().release_value_but_fixme_should_propagate_errors();
```
I think we do want to do this, but only when the display property actually gets changed from `none` to something else. Currently `play_or_cancel_animations_after_display_property_change()` gets called in `Element::inserted()` and `Element::removed()` and this branch will be taken in those cases when it shouldn't be.

If this branch definitely isn't supposed to be there then the method name should probably be changed.

Gingeh replied:
I've verified the behaviour locally, do you want me to add tests?

tcl3 commented on Tests/LibWeb/Text/input/css/initially-paused-animation.html:
```diff
@@ -0,0 +1,41 @@
+<!DOCTYPE html>
+<style>
+    body {
+        margin: 0;
+        display: flex;
+        justify-content: center;
+        align-items: center;
+        height: 100vh;
+        background-color: #f0f0f0;
+    }
```
This is a bit of a nitpick, but this styling doesn't seem necessary for the test. Getting rid of it also makes it esasier to see the text output.

tcl3 reviewed (commented):
I believe this change should give us at least +198 WPT subtests :tada:

Gingeh commented:
it's getting late here so I'll tinker more tomorrow

Gingeh commented:
I've modified the test to use `setTimeout` so the display type change doesn't happen during parsing, this then exposed an error caused by my code not calling pause on "should be paused" animations in `play_or_cancel_animations_after_display_property_change`. I've now fixed that bug.

Gingeh commented:
The code now also handles pseudo elements correctly

Gingeh commented:
test failure seems unrelated
